llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -2856,6 +2885,8 @@ void AddressSanitizer::initializeCallbacks(Module &M, const TargetLibraryInfo *T
       M.getOrInsertFunction(kAMDGPUAddressSharedName, IRB.getInt1Ty(), PtrTy);
   AMDGPUAddressPrivate =
       M.getOrInsertFunction(kAMDGPUAddressPrivateName, IRB.getInt1Ty(), PtrTy);
+
+  DormantAsanFlag = M.getOrInsertGlobal(kAsanDormancyVarName, IRB.getInt1Ty());
```

Comment:
I wonder if this will work if asan is built as a shared library?

Reply:
Hmmmm... as in, we make a .so file compiled with asan and try to link it against non asan intrumented code? This is a good question... time to go testing

Reply:
I also meant if asan itself is built as a shared library, not sure if this is supported on Linux.